File: CinemaReservations/Java/CinemaAllocationsDomain/src/main/java/com/sdd/cinemaallocations/NoMovieScreeningFound.java
package com.sdd.cinemaallocations;

import lombok.EqualsAndHashCode;
import lombok.ToString;
import lombok.Value;

import java.util.List;

@Value
@ToString(callSuper = true)
@EqualsAndHashCode(callSuper = true)
public class NoMovieScreeningFound extends SeatsAllocated {

    public NoMovieScreeningFound(int partyRequested, List<Seat> reservedSeats) {
        super(partyRequested, reservedSeats);
    }
}

File: CinemaReservations/CSharp/CinemaAllocations.Domain/NoMovieScreeningFound.cs
using System.Collections.Generic;

namespace CinemaAllocations.Domain
{
    public class NoMovieScreeningFound : SeatsAllocated
    {
        public NoMovieScreeningFound(int partyRequested) : base(new List<Seat>(), partyRequested)
        {
        }
    }
}